opentracing-spring-messaging/src/main/java/io/opentracing/contrib/spring/integration/messaging/OpenTracingChannelInterceptor.java

```diff
@@ -66,8 +67,9 @@ public OpenTracingChannelInterceptor(Tracer tracer) {
       spanBuilder.asChildOf(extractedContext);
     }
 
-    Span span = spanBuilder.startActive(true).span();
-
+    Span span = spanBuilder.start();
+    Scope scope = tracer.activateSpan(span);
```

Comment:
Do we have to activate the scope here? If no maybe we could just pass the span around.

Reply:
@pavolloffay I believe yes, if we want of the future spans created during the message handling to use the current span as parent.